### PR TITLE
:memo:(testing): add tests documentation

### PR DESCRIPTION
**Pull request by georgettica**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,5 +9,4 @@ Folow a simple workflow:
 * Follow the bot instructions
 
 This operator is build using the [operator-sdk](https://sdk.operatorframework.io)
-
-
+For a rough overview of the test please read [this document](TESTS.md)
```
Comment (RiRa12621) [inline CONTRIBUTING.md:12]:
```suggestion
For a rough overview of the tests please read [this document](TESTS.md)
```

File: TESTS.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+The tests suite here starts with a standard of gomock and ginkgo
+
+for a basic understanding of these refer to:
+- [ginkgo's website](https://onsi.github.io/ginkgo/)
+- [gomocks github page](https://github.com/golang/mock)
+
+The mocks we mostly use are generated using `make go-generate` which relies on `mockgen`.
+This means you won't need to write your own mocks,
+But know how to use the existing ones.
```
Comment (RiRa12621) [inline TESTS.md:9]:
```suggestion
but know how to use the existing ones.
```

File: TESTS.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+The next thing is to understand the structure of the mocks.
+Each test, which is wrapped in an `It` block.
+has usually one or more of these sections:
+- `BeforeEach` block: to set default values
+- `JustBeforeEach` block: to use the default values (which were set in the BeforeEach)
+
+In addition ti the specific tests, we set in the top of the test file two global `BeforeEach` and `JustBeforeEach` blocks.
```
Comment (RiRa12621) [inline TESTS.md:17]:
```suggestion
In addition to the specific tests, we set in the top of the test file two global `BeforeEach` and `JustBeforeEach` blocks.
```

**Comment by georgettica** [comment]
next time I will run a spelling on it, sry 😢

**Pull request merged**